src/main/kotlin/systems/ajax/malov/stockanalyzer/config/beanpostprocessor/LogExecutionTimeAnnotationBeanPostProcessor.kt

```diff
@@ -0,0 +1,82 @@
+package systems.ajax.malov.stockanalyzer.config.beanpostprocessor
+
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.config.BeanPostProcessor
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
+import org.springframework.stereotype.Component
+import java.lang.reflect.Method
+import java.lang.reflect.Proxy
+
+
+@ConditionalOnProperty(name = ["logExecutionTime.enabled"], havingValue = "true")
+@Component
+class LogExecutionTimeAnnotationBeanPostProcessor : BeanPostProcessor {
+
+    private val beanMap = HashMap<String, Pair<Class<*>, List<Method>>>()
```

Comment:
may be extract `Pair<Class<*>, List<Method>>` to a separate data class for readability?